python/pyiceberg/utils/datetime.py

```diff
@@ -61,15 +61,14 @@ def days_to_date(days: int) -> date:
     return EPOCH_DATE + timedelta(days)
 
 
-def time_to_micros(time_str: str) -> int:
+def time_str_to_micros(time_str: str) -> int:
     """Converts an ISO-8601 formatted time to microseconds from midnight."""
-    t = time.fromisoformat(time_str)
-    return (((t.hour * 60 + t.minute) * 60) + t.second) * 1_000_000 + t.microsecond
+    return time_to_micros(time.fromisoformat(time_str))
 
 
-def time_object_to_micros(t: time) -> int:
-    """Converts an datetime.time object to microseconds from midnight."""
-    return int(t.hour * 60 * 60 * 1e6 + t.minute * 60 * 1e6 + t.second * 1e6 + t.microsecond)
+def time_to_micros(t: time) -> int:
+    """Converts a datetime.time object to microseconds from midnight."""
+    return int((((t.hour * 60 * 60) + (t.minute * 60) + t.second) * 1e6) + t.microsecond)
```

Comment:
Can you make this use the same logic as the original?

```python
return (((t.hour * 60 + t.minute) * 60) + t.second) * 1_000_000 + t.microsecond
```

Needlessly changing logic that we know works always makes me paranoid.

Reply:
Oh, and IIRC the 1e6 converts the whole expression to float and back. We shouldn't do that.

Reply:
```suggestion
    return (((t.hour * 60 + t.minute) * 60) + t.second) * 1_000_000 + t.microsecond
```

> Oh, and IIRC the 1e6 converts the whole expression to float and back. We shouldn't do that.

This is correct:


```
>>> from datetime import time
>>> t = time(19, 25, 22)
>>> t
datetime.time(19, 25, 22)
>>> (((t.hour * 60 * 60) + (t.minute * 60) + t.second) * 1e6)
69922000000.0
>>> type(((t.hour * 60 * 60) + (t.minute * 60) + t.second) * 1e6)
<class 'float'>
>>> (((t.hour * 60 + t.minute) * 60) + t.second) * 1_000_000 + t.microsecond
69922000000
```